modules/versioning.nix

```
@@ -6,14 +6,43 @@ let
 
   # Sorted by increasing version numbers
   changes = [
-    # None yet
-    # {
-    #   version = "0.1";
-    #   condition = config.services.foo.enabled;
-    #   message = ''
-    #     demo message
-    #   '';
-    # }
+    {
+      version = "0.0.26";
+      condition = config.services.joinmarket.enable;
+      message = let
+        inherit (config.services.joinmarket) dataDir;
+      in ''
+        JoinMarket 0.8.0 moves from wrapped segwit wallets to native segwit wallets.
+
+        If you have an existing wrapped segwit wallet, you have to manually migrate
+        your funds to a new native segwit wallet.
+
+        To migrate, you first have to deploy the new JoinMarket version:
+        1. Set `nix-bitcoin.configVersion = "0.0.26";` in your configuration.nix
+        2. Deploy the new configuration
+
+        Then run the following on your nix-bitcoin node:
+        1. Move your wallet:
+           mv ${dataDir}/wallets/wallet.jmdat ${dataDir}/wallets/old.jmdat
+        2. Autogenerate a new p2wpkh wallet:
+           systemctl restart joinmarket
+        3. Transfer your funds manually by doing sweeps for each mixdepth:
+           jm-sendpayment -m <mixdepth> -N 0 old.jmdat 0 <destaddr>
+
+           Explanation of the options:
+           -m <mixdepth>: spend from given mixdepth.
+                          Repeat the command for every available mixdepth, e.g. `-m 0`, `-m 1`, ...
+           -N 0: don't coinjoin on this spend
+           old.jmdat: spend from old wallet
+           0: set amount to zero to do a sweep, i.e. transfer all funds at given mixdepth
+           <destaddr>: destination p2wpkh address from wallet.jmdat with mixdepth 0
+
+           You might want to time stagger these 4 transactions.
```

```suggestion
           You might want to time stagger these transactions.
```
The number of max. mixdepths is configurable and the default is 5 (= 5 transactions).